common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -165,7 +165,13 @@ trait Logging {
   }
 
   protected def withLogContext(context: java.util.HashMap[String, String])(body: => Unit): Unit = {
-    val threadContext = CloseableThreadContext.putAll(context)
+    // put into thread context only when structured logging is enabled
+    val updatedContext = if (Logging.isStructuredLoggingEnabled) {
+      context
+    } else {
+      new java.util.HashMap[String, String]()
```

Comment:
nit: shall we keep this to a val in object `Logging` instead of creating a new one for each log entry?